File: ascenso-backend/src/main/java/com/ais/ascensobackend/catalogo/application/services/interfaces/DestrezaService.java
package com.ais.ascensobackend.catalogo.application.services.interfaces;

import com.ais.ascensobackend.catalogo.application.dtos.DestrezaResumen;
import java.util.List;

public interface DestrezaService {

    List<DestrezaResumen> listar(Long grupoId);

    DestrezaResumen obtener(Long id);

    DestrezaResumen crear(Long grupoId, String nombre, String categoria);

    DestrezaResumen actualizar(Long id, String nombre, String categoria);

    void activar(Long id);

    void desactivar(Long id);
}

File: ascenso-backoffice/src/services/catalogo/DestrezaService.ts
import { apiClient } from '@/services/http/ApiClient'
import { API_ENDPOINTS } from '@/config/endpoints'
import type { Destreza, GuardarDestrezaRequest } from '@/types/catalogo'

class DestrezaService {
  listar(grupoId?: number) {
    return apiClient.get<Destreza[]>(API_ENDPOINTS.catalogo.destrezas, {
      params: grupoId ? { grupoId } : undefined,
    })
  }

  crear(request: GuardarDestrezaRequest) {
    return apiClient.post<Destreza>(API_ENDPOINTS.catalogo.destrezas, request)
  }

  actualizar(id: number, request: GuardarDestrezaRequest) {
    return apiClient.put<Destreza>(API_ENDPOINTS.catalogo.destrezaPorId(id), request)
  }

  activar(id: number) {
    return apiClient.post<void>(API_ENDPOINTS.catalogo.destrezaActivar(id))
  }

  desactivar(id: number) {
    return apiClient.post<void>(API_ENDPOINTS.catalogo.destrezaDesactivar(id))
  }
}

export const destrezaService = new DestrezaService()
